core/execution.go

```diff
@@ -100,6 +102,9 @@ func exec(env vm.Environment, caller vm.ContractRef, address, codeAddr *common.A
 		}
 	}
 	env.Transfer(from, to, value)
+	if !createAccount && env.Db().GetCodeHash(*codeAddr) == daoHash && value.Cmp(common.Big0) > 0 {
```

Comment:
Not really sure about this value comparison. Doesn't this actually block value transfers **into** the dao rather than stop funds leaking out of it?

Shouldn't a simpler comparison be to check for dao code match at the top of this method, if matches query the balance, and then query the balance after the end of execution. If the balance went down, set the flag?